lib/chart/some_chart/settings.ex

```diff
@@ -3,34 +3,33 @@ defmodule Chart.SomeChart.Settings do
   A chart settings.
   """
 
-  alias Chart.Internal.{
-    Axis,
-    Figure,
-    Grid,
-    Label,
-    MajorTicks,
-    MajorTicksText,
-    MinorTicks,
-    Plot,
-    Title
-  }
+  alias Chart.Internal.AxisLine.{Helpers, Label, MajorTicks, MajorTicksText, MinorTicks}
+  alias Chart.Internal.{AxisLine, Figure, GridLine, Plot, Title}
 
   def new() do
     %{}
     |> Figure.add()
     |> Title.add()
+    |> Title.set_text("Graph")
     |> Plot.add()
-    |> Grid.add()
+    |> GridLine.add()
     |> axis(:x_axis)
     |> axis(:y_axis)
+    |> Label.set_text(:x_axis, "Axis X")
+    |> Label.set_text(:y_axis, "Axis Y")
+    |> Label.set_placement(:y_axis, :top)
   end
 
-  def axis(settings, key) when is_map(settings) do
+  def axis(settings, key) when is_map(settings) and is_atom(key) do
     settings
-    |> Axis.add(key)
+    |> AxisLine.add(key)
     |> Label.add(key)
     |> MajorTicks.add(key)
     |> MajorTicksText.add(key)
     |> MinorTicks.add(key)
+    |> Helpers.recalculate_ticks_positions(key)
+    |> Helpers.recalculate_line(key)
+    |> Helpers.recalculate_label_position(key)
+    |> Helpers.recalculate_ticks_labels(key)
```

Comment:
Proc je to recalculate, kdyz by toto melo vytvorit rozumny default? Pokud je neco potreba vypocitat, tak proc to neni schovane v patricne fci `add`?

Reply:
Puvodne jsem to zacinal psat pro ty settry. Odtud nazev recalculate. Ano muze to byt v `add` opravim to.